pkg/controller/machinepool/openstackactuator.go

```diff
@@ -45,9 +46,10 @@ func NewOpenStackActuator(masterMachine *machinev1beta1.Machine, kubeClient clie
 		return nil, err
 	}
 	actuator := &OpenStackActuator{
-		logger:     logger,
-		osImage:    osImage,
-		kubeClient: kubeClient,
+		logger:                 logger,
+		osImage:                osImage,
+		kubeClient:             kubeClient,
+		trunkSupportDiscoverer: installosp.CheckNetworkExtensionAvailability,
```

Comment:
nit: the function can be scoped to only care about the Trunk extension, if that's all we're interested about:

```Go
type OpenStackActuator struct {
	// ...
	trunkSupportDiscoverer func(cloud string, opts *clientconfig.ClientOpts) (bool, error)
```

```Go
actuator := &OpenStackActuator{
	// ...
	trunkSupportDiscoverer: func(cloud string, opts *clientconfig.Clientopts) (bool, error) {
		return installosp.CheckNetworkExtensionAvailability(cloud, "trunk", opts)
	},
```

Reply:
I also think you should use what your team is used to! :)

Reply:
I do think the current approach is cleaner in terms of readability